src/components/RouteSchedule/RouteSchedule.js

```diff
@@ -0,0 +1,233 @@
+
+/**
+ * Returns a paded number (with leading 0 for integer < 10).
+ * @param {Number} number number.
+ */
+const pad = number => {
+  return `${number < 10 ? '0' : ''}${number}`;
+};
+
+/**
+ * Returns a 'hh:mm' string from a time.
+ * @param {Number} t time in milliseconds.
+ */
+const getTimeString = t => {
+  if (t === -1) {
+    return '';
+  }
+  const h = Math.floor(t / 36000000);
+  const m = Math.floor((t % 36000000) / 600000);
+  return `${pad(h)}:${pad(m)}`;
+};
+
+/**
+ * Returns a color class to display the delay.
+ * @param {Number} time Delay time in milliseconds.
+ */
+const getDelayString = t => {
+  const h = Math.floor(t / 3600000);
+  const m = Math.floor((t % 3600000) / 60000);
+  const s = Math.floor(((t % 3600000) % 60000) / 1000);
+
+  if (s === 0 && h === 0 && m === 0) {
+    return '0';
+  }
+  if (s === 0 && h === 0) {
+    return `${pad(m)}m`;
+  }
+  if (s === 0) {
+    return `${pad(h)}h${pad(m)}m`;
+  }
+  if (m === 0 && h === 0) {
+    return `${pad(s)}s`;
+  }
+  if (h === 0) {
+    return `${pad(m)}m${pad(s)}s`;
+  }
+  return `${pad(h)}h${pad(m)}m${pad(s)}s`;
+};
+
+/**
+ * Returns a color class to display the delay.
+ * @param {Number} time Delay time in milliseconds.
+ */
+const getDelayColor = time => {
+  const secs = Math.round(((time / 1800 / 2) * 3600) / 1000);
+  if (secs >= 3600) {
+    return 'dark-red';
+  }
+  if (secs >= 500) {
+    return 'middle-red';
+  }
+  if (secs >= 300) {
+    return 'light-red';
+  }
+  if (secs >= 180) {
+    return 'orange';
+  }
+  return 'green';
+};
+
+/**
+ * Returns an image for first, middle or last stations.
+ * @param {Number} index index of the station in the list.
+ * @param {Number} length Length of the stations list.
+ */
+const renderStationImg = (index, length) => {
+  if (index === 0) {
+    return (
+      <img
+        src={firstStation}
+        alt="routeScheduleLine"
+        className="rt-route-icon"
+      />
+    );
+  }
+  if (index === length - 1) {
+    return (
+      <img
+        src={lastStation}
+        alt="routeScheduleLine"
+        className="rt-route-icon"
+      />
+    );
+  }
+  return (
+    <img src={station} alt="routeScheduleLine" className="rt-route-icon" />
+  );
+};
+
+const propTypes = {
+  /**
+   * CSS class of the route schedule wrapper.
+   */
+  className: PropTypes.string,
+
+  /**
+   * Trajectory stations informations.
+   */
+  lineInfos: PropTypes.object,
+
+  /**
+   * HTML tabIndex attribute
+   */
+  stationTabIndex: PropTypes.number,
+
+  /**
+   * Render Header of the route scheduler.
+   */
+  renderHeader: PropTypes.func,
+
+  /**
+   * Render Body of the route scheduler.
+   */
+  renderStations: PropTypes.func,
+
+  /**
+   * Function triggered on station's click event.
+   */
+  onStationClick: PropTypes.func,
+};
+
+const defaultProps = {
+  className: 'rt-route-wrapper',
+  lineInfos: null,
+  stationTabIndex: 0,
+  renderHeader: lineInfos => {
+    return (
```

Comment:
I don't find this readable and I don't think `defaultProps` should be used for whole function components. That's not what they are made for...